### ISSUE TITLE
0.7.3: Tablet: Thunar always makes multi selection on left click

### ISSUE THREAD

**Issue by Narrat** [labwc/labwc#1985]
Since the labwc 0.7.3 release I notice a different behaviour with the thunar file manager and a tablet.
If I don't directly left click with the pen on a file or directory it will create a multi selection with a random starting point (most of the time top left in the file manager) like when creating a selection with pressed left mouse button. Trying to select multiple files the regular way doesn't work now.
![labwc_thunar](https://github.com/user-attachments/assets/00b4873a-c03d-41fb-ab09-a0425106d2c0)

After bisecting it, this change was introduced with fb57d61f3547d4826d47459b15292389b6a0838c

But I'm unsure if this is an intended change, something broken within thunar or gtk3 or misconfiguration on my side. Any pointers?

**Comment by johanmalm** [comment]
@jp7677 would you be happy to have a look a this one?

**Comment by jp7677** [comment]
I’m currently not in front of my machine, will look more closely at thunar soon.In the meantime, you can switch back to mouse emulation for tablets, see the tablet section in the documentation.On 15 Jul 2024, at 09:07, Johan Malm ***@***.***> wrote:﻿
@jp7677 would you be happy to have a look a this one?

**Comment by jp7677** [comment]
Looking more closely at this, Thunar draws a selection from the last reported pointer coordinate to the current tablet coordinate on stylus down. Since we "park" the pointer cursor at -1/-1 (https://github.com/labwc/labwc/blob/master/src/input/tablet.c#L233), the selection starts at upper-left. Marking the pointer cursor "left" on tablet events might fix this, but (see the comment in the linked source) this leads to other side effects. 

I tend to say that this should be addressed in Thunar, also considering that the GTK3-Demo "Icon View Basics" doesn't show a selection on stylus down. As stated above, switching to mouse emulation mitigates this.

**Comment by Narrat** [comment]
Thank you for looking into this.
Indeed, after setting mouse emulation active helped, when you first mentioned it. I hadn't realized that I changed behaviour, when I merged new config elements.
If needed I can set a Button for toggling. And maybe sometimes in the future this can be achieved via `WindowRules`?

Is here someone around that is involved with XFCE? I thought I had seen a name, but I'm not sure. Would be easier, before I go on the journey on recovering my old account for the bugtracker there... :D

**Comment by Consolatis** [comment]
The `-1,-1` pointer parking sounds like something that we should address at some point. Sending a `pointer_leave()` event sounds sensible if the client supports the tablet protocol.

**Comment by jp7677** [comment]
> The `-1,-1` pointer parking sounds like something that we should address at some point. Sending a `pointer_leave()` event sounds sensible if the client supports the tablet protocol.

Yes, I agree. That said, doing so will probably require some refactor work and lots of testing considering all the details in the cursor handling.

**Comment by Consolatis** [comment]
I have a feeling that we could just add something like ` && !tablet_has_surface_focus(surface)` to the condition in https://github.com/labwc/labwc/blob/master/src/input/cursor.c#L476. But I didn't actually look deeper into it.
